ogr/ogrsf_frmts/miramon/ogrmiramonlayer.cpp

```diff
@@ -0,0 +1,2613 @@
+                            MM_WRITTING_MODE, MMMap))
+            {
+                bValidFile = false;
+                return;
+            }
+            hMiraMonLayerPNT.bIsBeenInit = 0;
+
+            CPLDebug("MiraMon", "Initializing MiraMon arcs layer...");
+            if (MMInitLayer(&hMiraMonLayerARC, pszFilename, nMMVersion,
+                            nMMRecode, nMMLanguage, nMMMemoryRatio, nullptr,
+                            MM_WRITTING_MODE, MMMap))
+            {
+                bValidFile = false;
+                return;
+            }
+            hMiraMonLayerARC.bIsBeenInit = 0;
+
+            CPLDebug("MiraMon", "Initializing MiraMon polygons layer...");
+            if (MMInitLayer(&hMiraMonLayerPOL, pszFilename, nMMVersion,
+                            nMMRecode, nMMLanguage, nMMMemoryRatio, nullptr,
+                            MM_WRITTING_MODE, MMMap))
+            {
+                bValidFile = false;
+                return;
+            }
+            hMiraMonLayerPOL.bIsBeenInit = 0;
+
+            // Just in case that there is no geometry but some other
+            // information to get. A DBF will be generated
+            CPLDebug("MiraMon", "Initializing MiraMon only-ext-DBF layer...");
+            if (MMInitLayer(&hMiraMonLayerReadOrNonGeom, pszFilename,
+                            nMMVersion, nMMRecode, nMMLanguage, nMMMemoryRatio,
+                            nullptr, MM_WRITTING_MODE, nullptr))
+            {
+                bValidFile = false;
+                return;
+            }
+            hMiraMonLayerPOL.bIsBeenInit = 0;
+
+            // This helps the map to be created
+            //GetLayerDefn()->SetName(hMiraMonLayerPNT.pszSrcLayerName);
+            poFeatureDefn->SetName(hMiraMonLayerPNT.pszSrcLayerName);
+        }
+
+        // Saving the HRS in the layer structure
+        if (poSRS)
+        {
+            if (poSRS->GetAuthorityName(nullptr) &&
+                EQUAL(poSRS->GetAuthorityName(nullptr), "EPSG"))
+            {
+                CPLDebug("MiraMon", "Setting EPSG code %s",
+                         poSRS->GetAuthorityCode(nullptr));
+                hMiraMonLayerPNT.pSRS =
+                    CPLStrdup(poSRS->GetAuthorityCode(nullptr));
+                hMiraMonLayerARC.pSRS =
+                    CPLStrdup(poSRS->GetAuthorityCode(nullptr));
+                hMiraMonLayerPOL.pSRS =
+                    CPLStrdup(poSRS->GetAuthorityCode(nullptr));
+            }
+        }
+    }
+    else
+    {
+        if (m_fp == nullptr)
+        {
+            bValidFile = false;
+            return;
+        }
+
+        /* ------------------------------------------------------------------*/
+        /*      Read the header.                                             */
+        /* ------------------------------------------------------------------*/
+        if (!STARTS_WITH(pszFilename, "/vsistdout"))
```

Comment:
test to remove

Reply:
I don't know how to test that. I tryed at sometime but... any suggestions?

Reply:
I was just saying you should just remove the ``if (!STARTS_WITH(pszFilename, "/vsistdout"))`` line. There is no need to worry specifically about /vsistdout/

Reply:
Ah, ok, i copyied it from the model I used long time ago. Perhaps here has no sense to use that.